shellcheck.yml

```diff
@@ -0,0 +1,43 @@
+name: Shellcheck Lint
+
+on:
+  push:
+    paths:
+      # Run workflow on push to file pihole_adlist_tool
+      - pihole_adlist_tool
+
+  pull_request:
+    branches:
+      # Run workflow on pull request to master
+      - master
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Shellcheck Lint
+
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+
+    steps:
+      # Required to access files of this repository
+      - uses: actions/checkout@v2
+
+      # Download Shellcheck and add it to the workflow path
+      - name: Download Shellcheck
+        run: |
+          wget -qO- "https://github.com/koalaman/shellcheck/releases/download/stable/shellcheck-stable.linux.x86_64.tar.xz" | tar -xJv
+          chmod +x shellcheck-stable/shellcheck
+      # Verify that Shellcheck can be executed
+      - name: Check Shellcheck Version
+        run: |
+          shellcheck-stable/shellcheck --version
```

Comment:
Don't we need a if clause here to test if there is any output?

Reply:
No, we don't need an `if` clause or something like that. Have a look at the link to my fork, where you can see the output from one failing example (I didn't ignore SC2166 for this run):
https://github.com/thomasmerz/pihole_adlist_tool/runs/4999213693?check_suite_focus=true